packages/pigeon/lib/swift_generator.dart

```diff
@@ -268,7 +274,8 @@ import FlutterMacOS
         final _SwiftFunctionComponents components =
             _SwiftFunctionComponents.fromMethod(func);
 
-        final String channelName = makeChannelName(api, func);
+        final String channelName =
+            makeChannelName(api, func, generatorOptions.packageName!);
```

Comment:
Can we make the fields non-nullable so we're not force-unwrapping them everywhere, and instead ensure that they are non-null at the point where we populate them (which is where that logic actually lives)? Currently it's very hard to tell if/why this is safe because the unwrapping is so far from the logic that populates it.

Reply:
I'll look at it, in general I'm trying to keep all pigeonOptions nullable to match all other inputs and allow for consistent logic on the `merge` methods.

Reply:
Maybe we should pass the package name as a separate non-nullable argument into the generators then, rather than relying on the options to pass it?

Fundamentally it seems the issue here is that we're using the same object for user-facing options (where it's an optional parameter) and internal implementation (where have some value is required).

Reply:
I can change the non user faces options, if that makes the most sense